chart/templates/post-delete-hook-job.yaml

```diff
@@ -0,0 +1,47 @@
+{{- if .Values.postDelete.enabled }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Values.postDelete.name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "rancher.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": post-delete
+    "helm.sh/hook-weight": "3"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  backoffLimit: 3
+  template:
+    metadata:
+      name: {{ .Values.postDelete.name }}
+      labels:
+        heritage: {{ .Release.Service }}
+        release: {{ .Release.Name }}
+    spec:
+      serviceAccountName: {{ .Values.postDelete.name }}
+      restartPolicy: OnFailure
+      containers:
+        - name: {{ .Values.postDelete.name }}
+          image: "{{ .Values.systemDefaultRegistry }}{{ .Values.postDelete.image.repository }}:{{ .Values.postDelete.image.tag }}"
+          imagePullPolicy: IfNotPresent
+          securityContext:
+            runAsUser: 0
+          command:
+            - /script/post-delete-hook.sh
+          volumeMounts:
+            - mountPath: /script
+              name: config-volume
+          env:
+            - name: namespaces
+              value: {{ .Values.postDelete.namespaceList | join " " | quote }}
+            - name: timeout
+              value: {{ .Values.postDelete.timeout | quote }}
+            - name: ignoreTimeoutError
+              value: {{ .Values.postDelete.ignoreTimeoutError | quote }}
```

Comment:
environment variables that come from outside the bash script should be capitalized, e.g. `NAMESPACES`.

Also, are the `| quote` steps necessary?

Reply:
> environment variables that come from outside the bash script should be capitalized, e.g. NAMESPACES. 

This is fixed. 

> Also, are the | quote steps necessary?

THe `helm uninstall rancher -n cattle-system` will throw the following error if there are no quotes around those env vars

`>  helm uninstall rancher -n cattle-system
Error: uninstallation completed with 1 error(s): warning: Hook post-delete rancher/templates/post-delete-hook-job.yaml failed: Job in version "v1" cannot be handled as a Job: v1.Job.Spec: v1.JobSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.v1.Container.Env: []v1.EnvVar: v1.EnvVar.Value: ReadString: expects " or n, but found 1, error found in #10 byte of ...|,"value":120},{"name|..., bigger context ...|cher-operator-system"},{"name":"TIMEOUT","value":120},{"name":"IGNORETIMEOUTERROR","value":false}],"|...
`